nav2_behavior_tree/include/nav2_behavior_tree/plugins/decorator/is_path_longer_on_approach.hpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2022 Neobotix GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__DECORATOR__IS_PATH_LONGER_ON_APPROACH_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__DECORATOR__IS_PATH_LONGER_ON_APPROACH_HPP_
+
+#include <string>
+#include <memory>
+#include <limits>
+
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "behaviortree_cpp_v3/decorator_node.h"
+
+namespace nav2_behavior_tree
+{
+
+/**
+ * @brief A BT::DecoratorNode that ticks its child everytime when the length of
+ * the new path is smaller than the old one by the length given by the user.
+ */
+class IsPathLongerOnApproach : public BT::DecoratorNode
+{
+public:
+  /**
+   * @brief A constructor for nav2_behavior_tree::IsPathLongerOnApproach
+   * @param name Name for the XML tag for this node
+   * @param conf BT node configuration
+   */
+  IsPathLongerOnApproach(
+    const std::string & name,
+    const BT::NodeConfiguration & conf);
+
+  /**
+   * @brief Creates list of BT ports
+   * @return BT::PortsList Containing node-specific ports
+   */
+  static BT::PortsList providedPorts()
+  {
+    return {
+      BT::InputPort<nav_msgs::msg::Path>("new_path", "New Planned Path"),
+      BT::InputPort<nav_msgs::msg::Path>("old_path", "Old Planned Path"),
```

Comment:
Remove this port, there is no reason for this, it should be stored internally. Then the single remaining input port should be `path` like every other port API

Reply:
I added it mainly for the purpose of testing the failure case, without having an `old_path` it is highly impossible to test the failure case.

Reply:
Why not just run the tree twice? I don't think you should be adding things like that purely for testing. There's got to be work arounds for testing like making a new BT node that derives from this that you can set the path before executing to have something in it beforehand.